app/services/transaction.service.js

```diff
@@ -148,6 +148,15 @@ function getStatusCodeForError (err, response) {
   return status
 }
 
+function handleErrorForFailedSearch (err, response) {
+  const code = (response || {}).statusCode || (err || {}).errorCode
```

Comment:
Not sure about this construct… `{}.errorCode` is never going to exist. Did you perhaps mean to do something like `if (err.errorCode) { …`?